libs/chain/include/chain/transaction.hpp

```diff
@@ -137,6 +137,8 @@ class Transaction
   bool Verify();
   bool IsVerified() const;
   bool IsSignedByFromAddress() const;
+  void SetFromSync();
```

Comment:
I think these names are a little vague. Maybe invert the logic and use `SetRecent(bool recent = true)` and `IsRecent()` - this will more closely match the other API too I guess

Reply:
One thing I didn't like about this inversion was that it requires calling `SetRecent` everywhere a recent tx may be received, while `SetFromSync` is only needed after a sync, and all other tx handling sites, either real or imaginary, can be left intact, unaware of this bit.

Reply:
Sure okay that is fair - Can we be more explicit then and go with. `SetFromSubtreeSync` etc? Thoughts?

Reply:
Changed to .*FromSubtreeSync, don't have more descriptive ideas either.